scraper/doctolib/doctolib.py

```diff
@@ -85,6 +95,45 @@ def fetch(self, rdv_site_web: str, start_date: str) -> Optional[str]:
 
         return slots.get('next_slot')
 
+    def centre_iterator(self, max_page=1000):
+
+        page = 1
+        centres = True
+
+        while centres and page <= max_page:
+
+            url_recherche = BASE_URL + RECHERCHE_URL + "page={}&".format(page) + PARAMETRES
+
+            response = self._client.get(url_recherche, headers=DOCTOLIB_HEADERS)
+
+            try:
+                response.raise_for_status()
+                soup = BeautifulSoup(response.content, "html.parser")
+                centres = soup.find_all("div", {"class": "dl-search-result"})
+
+                for _centre in centres:
+
+                    centre = {}
+
+                    centre['gid'] = _centre.get('id')[14:]
+
+                    nom = _centre.select_one(".dl-search-result-name").getText()
+                    centre['nom'] = nom
+
+                    rdv_site_web = BASE_URL + _centre.select_one('a[data-analytics-event-action="bookAppointmentButton"]')['href']
+                    centre['rdv_site_web'] = rdv_site_web
+
+
+                    adresse = _centre.select_one('.dl-text.dl-text-body.dl-text-s').getText().split(' ')
+                    centre["com_insee"] = cp_to_insee(_recherche_cp(adresse))
+
+                    yield centre
+
+            except Exception as e:
+                logger.error(f"erreur lors du traitement de recherche sur Doctolib {str(e)}")
```

Comment:
why not tu peux itérer plus haut

Reply:
qu'est ce que tu entends par itérer plus haut ?